File: Array/array.py
def INSERT(M,Arr,Ci):
  for i in range(len(M)):
    if M[i]=="+":
      if len(Arr)==0:
        Ci=0
      else :
        Ci+=1
      Arr.insert(Ci,M[i+1])
  return Ci
  
def TRAVERSE_FRONT(M,Arr,Ci):
  Ci=0
  Nip=0;
  for i in range(len(M)):
    if M[i]=="N":
      Ci+=1
    if M[i]=='M':
      if M[i+1]=='n':
        np=len(Arr)-1
      elif M[i+1]=='P':
        np=Ci-1
      elif M[i+1]=='N':
        np=Ci+1
      else:
        np=int(M[i+1])
      if np<Ci:
        Arr.insert(np,Arr[Ci])
        del Arr[Ci+1]
        Ci=int(np)
      else:
        np=np+1
        Arr.insert(np,Arr[Ci])
        del Arr[Ci]
        Ci=np-1
  return Ci
  
def TRAVERSE_REAR_DELETE(M,Arr,Ci):
  Ci=len(arr)-1
  for i in range(len(M)):
      if M[i]=="P":
        Ci-=1
      if M[i]=='-':
        del Arr[Ci]
        Ci=0 
  return Ci

def DELETE(Arr,Ci):
  if(Ci==(len(Arr)-1)):
    del Arr[Ci]
    Ci=0
  else:
    del Arr[Ci]
  return Ci

def GET_DATA(arr,ci):
  print('Return',arr[ci])

def REPLACE(M,Arr,Ci):
  Arr.insert(Ci,M[1])
  del Arr[Ci+1]
  return Ci
  
def EMPTY(arr):
  arr.clear()
  print("empty array")

def MOVE(Arr,Np,Ci):
  if Np<Ci:
    Arr.insert(Np,Arr[Ci])
    del Arr[Ci+1]
    Ci=int(Np)
  else:
    Np=Np+1
    Arr.insert(Np,Arr[Ci])
    del Arr[Ci]
    Ci=Np-1
  return Ci

def DATA_COUNT(Arr):
  return len(Arr)

def UPPER(Arr,Ci):
  return Arr[Ci].upper()

def LOWER(Arr,Ci):
  return Arr[Ci].lower()

def pt():
  for i in range(len(arr)):
      if i==ci:
        print("(%s) "%arr[i],end='')
      else:
        print("%s "%arr[i],end='')
  print()

def V():
    print("----------MENU----------\n")
    print("INSERT               : +(data)\n")
    print("TRAVERSE_FRONT       : <(N)\n")
    print("TRAVERSE_REAR        : >(P)\n")
    print("DELETE               : -\n")
    print("DELETE THE LAST      : > -(띄어쓰세요)\n")
    print("GET_DATA             : @\n")
    print("REPLACE              : =(data)(붙여쓰세요)\n")
    print("EMPTY                : E\n")
    print("TRAVERSE_FRONT&&MOVE : < M(POSITION)\n")
    print("<과 M은 띄어쓰고 M과 POSITION은 붙여쓰세요\n")
    print("MOVE                 : M(POSITION)(붙여쓰세요)\n")
    print("MOVE END             : Mn(붙여쓰세요)\n")
    print("MOVE PREV            : Mp(붙여쓰세요)\n")
    print("MOVE NEXT            : MN(붙여쓰세요)\n")
    print("PRINT                : L\n")
    print("VIEW                 : V\n")
    print("UPPER CASE           : U\n")
    print("LOWER CASE           : l\n")
    print("END                  : B\n")
arr=[]
ci=0;
V()
while True:
  m=input()
  if m[0]=="+":
    ci=INSERT(m,arr,ci)
    pt()
  elif m[0]=="<":
    ci=TRAVERSE_FRONT(m,arr,ci)
    pt()
  elif m[0]==">":
    ci=TRAVERSE_REAR_DELETE(m,arr,ci)
    pt()
  if m[0]=='-':
    ci=DELETE(arr,ci)
    pt()
  if m[0]=='@':
    GET_DATA(arr,ci)
  if m[0]=='=':
    ci=REPLACE(m,arr,ci)
    pt()
  if m[0]=='E':
    EMPTY(arr)
  if m[0]=='M':
    if m[1]=='n':
      np=DATA_COUNT(arr)-1
    elif m[1]=='P':
      np=ci-1
    elif m[1]=='N':
      np=ci+1
    else:
      np=int(m[1])
    ci=MOVE(arr,np,ci)
    pt()
  if m[0]=='L':
    pt()
  if m[0]=='V':
    V()
  if m[0]=='U':
    arr[ci]=UPPER(arr,ci)
    pt()
  if m[0]=='l':
    arr[ci]=LOWER(arr,ci)
    pt()
  if m[0]=='B':
    break;

File: Array/array.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
void print(char arr[], int cur){
  int j;
  for (j = 0; j < strlen(arr); j++) {
          if (j == cur) { printf("(%c) ", arr[j]); }
          else { printf("%c ", arr[j]); }
          }
}
int insert(char arr[], char m[], int cur) {
	int i,k;
    for(i=0;i<strlen(m);i++){
            if(m[i]=='+'){
              cur+=1;
              for (k = strlen(arr); k > cur; k--) {
              arr[k] = arr[k - 1];
              }
              arr[cur] = m[i+1];
            }
          }
    return cur;
}

int traverse_front(char arr[], char m[], int cur) {
    cur = 0;
    int nc = 0;
    int i;
    for(i=0;i<strlen(m);i++){
      if(m[i]=='N'){
        nc+=1;}
    }
    cur = cur + nc;
    return cur;
}

int traverse_rear(char arr[], char m[], int cur) {
    cur = strlen(arr) - 1;
    int nc = 0;
    int i;
    for(i=0;i<strlen(m);i++){
      if(m[i]=='P'){
        nc+=1;}
    }
    cur = cur - nc;
    return cur;
}

int deletethelast(char arr[], int cur) {
    int i;
    cur = strlen(arr) - 1;
    for (i = cur; i < strlen(arr); i++) {
        arr[i] = arr[i + 1];
    }
    cur = 0;
    return cur;
}

int Delete(char arr[], int cur) {
    int i;
    for (i = cur; i < strlen(arr); i++) {
        arr[i] = arr[i + 1];
    }
    if (cur == strlen(arr)){
        cur = 0;
    }
    return cur;
}

int get_data(char arr[], int cur) {
    printf("Return %c", arr[cur]);
    return cur;
}

int replace(char arr[], char m[], int cur) {
    arr[cur] = m[1];
    return cur;
}

int empty(char arr[], int cur) {
    memset(arr, 0, strlen(arr));
    printf("empty array");
    cur = -1;
    return cur;
}

int move(char arr[], int new_position, int cur) {
    char carry;
    int i;
    carry = arr[cur];
    if (cur < new_position) {
        for (i = cur + 1; i <= new_position; i++) {
            arr[i - 1] = arr[i];
        }
    }
    if (cur > new_position) {
        for (i = cur - 1; i >= new_position; i--) {
            arr[i + 1] = arr[i];
        }
    }
    cur = new_position;
    arr[cur] = carry;
    return cur;
}

int data_count(char arr[]) {
    int n;
    n = strlen(arr) - 1;
    return n;
}

void view_menu() {
    printf("----------MENU----------\n");
    printf("INSERT               : +(data)\n");
    printf("TRAVERSE_FRONT       : <(N)\n"); 
    printf("TRAVERSE_REAR        : >(P)\n");
    printf("DELETE               : -\n");
    printf("DELETE THE LAST      : > -(띄어쓰세요)\n");
    printf("GET_DATA             : @\n");
    printf("REPLACE              : =(data)(붙여쓰세요)\n");
    printf("EMPTY                : E\n");
    printf("TRAVERSE_FRONT&&MOVE : < M(POSITION)\n");
    printf("<과 M은 띄어쓰고 M과 POSITION은 붙여쓰세요\n");
    printf("MOVE                 : M(POSITION)(붙여쓰세요)\n");
    printf("MOVE END             : Mn(붙여쓰세요)\n");
    printf("MOVE PREV            : Mp(붙여쓰세요)\n");
    printf("MOVE NEXT            : MN(붙여쓰세요)\n");
    printf("PRINT                : L\n");
    printf("VIEW                 : V\n");
    printf("UPPER CASE           : U\n");
    printf("LOWER CASE           : l\n");
    printf("END                  : B\n");
}

void upper(char arr[], int cur) {
    arr[cur] -= 32;
    print(arr,cur);
}

void Lower(char arr[], int cur) {
    arr[cur] += 32;
    print(arr,cur);
}

int main() {
    char arr[30] = {};
    int cur = -1;
    view_menu();

    while (1) {
    	char m[30]={};
        gets(m);
        if (m[0] == '+') {
            cur = insert(arr, m, cur);
            print(arr,cur);
        }
        if (m[0]== '<') {
          
          if(m[2]=='M'){
            int np;
            np = atoi(&m[3]);
            cur = traverse_front(arr, m, cur);
            cur = move(arr, np, cur);
            print(arr,cur);
          }
          else{
            cur = traverse_front(arr, m, cur);
            print(arr,cur);
          }
        }
        if (m[0] == '>') {
          if(m[2]=='-'){cur = deletethelast(arr, cur);}
          else {cur = traverse_rear(arr, m,cur);}
          print(arr,cur);
            }
        if (m[0] == '-') {
            cur = Delete(arr, cur);
            print(arr,cur);
        }
        if (m[0] == '@') {
            cur = get_data(arr, cur);
        }
        if (m[0] == '=') {
            cur = replace(arr, m, cur);
            print(arr,cur);
        }
        if (m[0] == 'E') {
            cur = empty(arr, cur);
        }
        if (m[0] == 'M') {
          int np;
            if (m[1]=='n') {
                np = data_count(arr);
                cur = move(arr, np, cur);
            }
            else if (m[1]=='P') {
                np = cur - 1;
                cur = move(arr, np, cur);
            }
            else if (m[1]=='N') {
                np = cur + 1;
                cur = move(arr, np, cur);
            }
            else {
                np = atoi(&m[2]);
                cur = move(arr, np, cur);
            }
          print(arr,cur);
        }
        if (m[0] == 'L') {
            print(arr, cur);
        }
        if (m[0] == 'V') {
            view_menu();
        }
        if (m[0] == 'U') {
            upper(arr, cur);
        }
        if (m[0] == 'l') {
            Lower(arr, cur);
        }
        if (m[0]=='B'){
        	break;
		}
        }
	}
